### ISSUE TITLE
[1.2.0 Beta 2] poller.php causes dumped core

### ISSUE THREAD

**Issue by mortenstevens** [Cacti/cacti#2160]
Testing cacti-1.2.0 on the latest Fedora (rawhide) dev branch with MariaDB 10.3 and PHP 7.3 rc5. 

Nov 09 17:40:02 IMTSYSFC010.de.imt-systems.net systemd-coredump[6921]: Process 6898 (php) of user 48 dumped core.

                                                                       Stack trace of thread 6898:
                                                                       #0  0x000055cf3615bbee _efree (php)
                                                                       #1  0x00007f87a9842d3b n/a (mysqlnd.so)
                                                                       #2  0x00007f87a983ddd3 n/a (mysqlnd.so)
                                                                       #3  0x00007f87a9842455 n/a (mysqlnd.so)
                                                                       #4  0x00007f87a9840d45 n/a (mysqlnd.so)
                                                                       #5  0x00007f87a983ec88 n/a (mysqlnd.so)
                                                                       #6  0x00007f87a983edac n/a (mysqlnd.so)
                                                                       #7  0x00007f87a912fb88 n/a (pdo_mysql.so)
                                                                       #8  0x00007f87a98182b9 n/a (pdo.so)
                                                                       #9  0x00007f87a981c05e n/a (pdo.so)
                                                                       #10 0x000055cf36193d36 n/a (php)
                                                                       #11 0x000055cf361909bb zend_hash_graceful_reverse_destroy (php)
                                                                       #12 0x000055cf36180552 n/a (php)
                                                                       #13 0x000055cf3611ff22 php_module_shutdown (php)
                                                                       #14 0x000055cf35fe7644 n/a (php)
                                                                       #15 0x00007f87ad18d413 __libc_start_main (libc.so.6)
                                                                       #16 0x000055cf35fe776e _start (php)

Any ideas? I'm not sure if it's a cacti or PHP 7.3 bug.

Best regards,

Morten

**Comment by pautiina** [comment]
Cacti don't support php 7.3 and php 7.2.

**Comment by mortenstevens** [comment]
@pautiina 

cacti-1.1.38 works fine on PHP 7.2 and after some testing with cacti-1.2.0 beta 2 on PHP 7.3 it looks all very good. Overall, I think the latest cacti release should support PHP 7.2 and 7.3.

**Comment by netniV** [comment]
Actually 1.1.38 has numerous issues with 7.2 due to breaking changes. It doesn’t cause a dump though. 7.3 is released now? Haven’t tested against that.

**Comment by netniV** [comment]
The stack suggests a PHP/MySQL or MariaDB issue.

**Comment by mortenstevens** [comment]
@netniV 

Yes, it's probably not a cacti bug. I'm just reporting this because I saw this stack only with cacti on my Fedora test setup. Do you plan to support PHP 7.2 and 7.3 with the upcoming 1.2 release? I think it's really important to support PHP 7.2+ because many distros have already switched to PHP 7.2.

**Comment by netniV** [comment]
7.2 is supported on 1.2

**Comment by mortenstevens** [comment]
@cigamit 
@netniV 

PHP 7.3.0 has been released. http://php.net/ChangeLog-7.php#7.3.0

This issue also affects cacti 1.1.38 and the latest cacti 1.2 beta. I agree with you that it is very likely a PHP 7.3 bug. But I'm not able to reproduce this with other php based applications using php-mysqlnd. In the meantime maybe there is a workaround that we can use for cacti? (until this PHP bug is fixed)

**Comment by netniV** [comment]
I’m not even sure what the PHP bug is tbh but I will take a look this weekend and see if I can spot anything

**Comment by netniV** [comment]
Unfortunately, I didn't get time this weekend and with a server migration/go live this week, I'm going to be pretty tied up.  Hopefully I will have time next week.

**Comment by mortenstevens** [comment]
Update: This issue seems to be fixed with the latest PHP 7.3.2~RC1.

**Comment by netniV** [comment]
Thanks Morten, I never got around to testing this.

**Comment by cigamit** [comment]
@mortenstevens, @netniV, @paulgevers, @DavidLiedke, 

Ditto.  Bleeding edge is like running Cacti 1.2.0 or 1.2.1.  We hope 1.2.2 is our next stable stable before we start hammering away at the 100 or so feature requests.  Stay tuned.
